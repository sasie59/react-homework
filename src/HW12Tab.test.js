import React from 'react';
import { render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import HW from './HW12Tab';

test('should HW12Tab', () => {
  const { container } = render(<HW />);
  expect(container).toMatchSnapshot();
});

test('should click different div will appear different text', () => {
  const { container } = render(<HW />);
  expect(container).toMatchSnapshot();
});

test('should click tab2', () => {
  const { container } = render(<HW />);
  const tab2 = container.querySelector('div[key="1"]');
  userEvent.click(tab2);
  expect(container).toMatchSnapshot();

});



