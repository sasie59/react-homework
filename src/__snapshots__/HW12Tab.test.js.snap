// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should HW12Tab 1`] = `
<div>
  <div>
    <h1>
      HW12Tab
    </h1>
    <h1>
      標籤
    </h1>
    <div
      style="display: inline-block; width: 50px; height: 25px; border: 1px solid black;"
    >
      tab1
    </div>
    <div
      style="display: inline-block; width: 50px; height: 25px; border: 1px solid black;"
    >
      tab2
    </div>
    <div
      style="display: inline-block; width: 50px; height: 25px; border: 1px solid black;"
    >
      tab3
    </div>
    <div
      style="display: inline-block; width: 50px; height: 25px; border: 1px solid black;"
    >
      tab4
    </div>
    <div
      style="display: inline-block; width: 50px; height: 25px; border: 1px solid black;"
    >
      tab5
    </div>
    <br />
    lorem 1
  </div>
</div>
`;

exports[`should click different div will appear different text 1`] = `
<div>
  <div>
    <h1>
      HW12Tab
    </h1>
    <h1>
      標籤
    </h1>
    <div
      style="display: inline-block; width: 50px; height: 25px; border: 1px solid black;"
    >
      tab1
    </div>
    <div
      style="display: inline-block; width: 50px; height: 25px; border: 1px solid black;"
    >
      tab2
    </div>
    <div
      style="display: inline-block; width: 50px; height: 25px; border: 1px solid black;"
    >
      tab3
    </div>
    <div
      style="display: inline-block; width: 50px; height: 25px; border: 1px solid black;"
    >
      tab4
    </div>
    <div
      style="display: inline-block; width: 50px; height: 25px; border: 1px solid black;"
    >
      tab5
    </div>
    <br />
    lorem 1
  </div>
</div>
`;

exports[`should click tab2 1`] = `
<div>
  <div>
    <h1>
      HW12Tab
    </h1>
    <h1>
      標籤
    </h1>
    <div
      style="display: inline-block; width: 50px; height: 25px; border: 1px solid black;"
    >
      tab1
    </div>
    <div
      style="display: inline-block; width: 50px; height: 25px; border: 1px solid black;"
    >
      tab2
    </div>
    <div
      style="display: inline-block; width: 50px; height: 25px; border: 1px solid black;"
    >
      tab3
    </div>
    <div
      style="display: inline-block; width: 50px; height: 25px; border: 1px solid black;"
    >
      tab4
    </div>
    <div
      style="display: inline-block; width: 50px; height: 25px; border: 1px solid black;"
    >
      tab5
    </div>
    <br />
    lorem 1
  </div>
</div>
`;
